File: 2020/day6/day6.exs
File.read!("input.txt")
|> String.split("\n\n")
|> Enum.sum_by(fn group ->
  people = String.split(group) |> Enum.map(&String.graphemes/1)

  Enum.reduce(people, [], fn person, acc ->
    Enum.reduce(person, acc, fn q, acc ->
      if Enum.all?(people, fn p -> q not in acc and q in p end), do: [q | acc], else: acc
    end)
  end)
  |> Enum.count()
end)
|> IO.inspect()

File: 2022/day6/day6.py
with open("day6/input.txt", "r") as f:
    input = f.read()


def calculate(count):
    for x in range(len(input)):
        sliced = input[x : x + count]
        if len(set(sliced)) == count:
            return x + count


print(calculate(4), calculate(14))
